jwala-services/src/main/java/com/cerner/jwala/service/media/impl/MediaServiceImpl.java

```diff
@@ -85,8 +90,20 @@ public JpaMedia create(final Map<String, String> mediaDataMap, final Map<String,
     @Transactional
     public void remove(final String name) {
         final JpaMedia jpaMedia = mediaDao.find(name);
-        repositoryService.delete(jpaMedia.getLocalPath().getFileName().toString());
+        checkForExistingAssociationsBeforeRemove(name);
```

Comment:
I would make checkForExistingAssociationsBeforeRemove(name) return a boolean and throw an exception if true[may be rename to associationExists(name)], else remove it. This way we can reuse this method